src/index.ts

```diff
@@ -1,139 +1,115 @@
-                    );
-                    break;
+import { createDeployment, createDevelopmentServer, createNewContentBlock, loginUser, logoutUser } from './commands';
+import { getValidInstanceUrl } from './utils';
+import pkg from '../package.json';
+
+const cli = cac(pkg.name.split('/')[1]);
+
+cli.command('login [instanceUrl]', 'log in to a Frontify instance')
+    .option('-i, --instance [instanceUrl]', '[string] url of the Frontify instance')
+    .option('-p, --port <port>', '[number] port for the oauth service', {
+        default: process.env.PORT || 5600,
+    })
+    .action(async (instanceUrl: string, options) => {
+        let promptedInstanceUrl: string | null = null;
+
+        if (!instanceUrl && !options.instance && !process.env.INSTANCE_URL) {
+            const { value } = await prompts({
+                type: 'text',
+                name: 'value',
+                message: 'Frontify Instance URL',
+                initial: 'instanceName.frontify.com',
+                validate: (value: string) => (value === '' ? 'You need to enter a URL.' : true),
+            });
+
+            if (!value) {
+                exit(0);
             }
-            break;
 
-        case 'login':
-            const instanceUrl = getValidInstanceUrl(parseArgs.instance || process.env.INSTANCE_URL);
-            await loginUser(instanceUrl, parseArgs[Argument.Port]);
-            break;
+            promptedInstanceUrl = value;
+        }
+
+        const parsedInstanceUrl = getValidInstanceUrl(
+            instanceUrl ?? options.instance ?? process.env.INSTANCE_URL ?? promptedInstanceUrl
+        );
+
+        await loginUser(parsedInstanceUrl, options.port);
+    });
+
+cli.command('logout', 'log out of an instance').action(logoutUser);
 
-        case 'logout':
-            logoutUser();
-            exit(1);
+/**
+ * @deprecated `block serve` and `theme serve` will be removed in version 4.0 in favour of `serve`
+ */
+for (const appType of ['block', 'theme']) {
+    cli.command(`[root] ${appType} serve`, `serve the ${appType} locally`)
+        .alias(`${appType} dev`)
+        .option('-e, --entryPath, --entry-path <entryPath>', `[string] path to the ${appType} entry file`, {
+            default: join('src', 'index.tsx'),
+        })
+        .option('--port <port>', '[number] specify port', {
+            default: process.env.PORT || 5600,
+        })
+        .action(async (_args, options) => {
+            await createDevelopmentServer(options.entryPath, options.port);
+        });
+}
+
+cli.command('serve', 'serve the app locally')
+    .alias('dev')
+    .option('-e, --entryPath, --entry-path <entryPath>', '[string] path to the entry file', {
+        default: join('src', 'index.ts'),
+    })
+    .option('--port <port>', '[number] specify port', {
+        default: process.env.PORT || 5600,
+    })
+    .action(async (_args, options) => {
+        await createDevelopmentServer(options.entryPath, options.port);
```

Comment:
this fails fore me when calling it without the `entryPath` parameter:
<img width="710" alt="Screenshot 2022-10-24 at 09 08 24" src="https://user-images.githubusercontent.com/154760/197467859-ca8e4fdc-513b-41a3-bb69-45590e7f8d22.png">